### DISCUSSION TITLE
MAILER_EMAIL not working

### DISCUSSION THREAD

**Discussion by wheeleran** [plausible/analytics#2921] [Self-Hosted Support]
There are a number of tickets about this, some marked solved without any details about HOW it was solved. I've set MAILER_EMAIL to a custom email address and it's not working. Plausible self hosted on a synology NAS, docker-compose. Here's my bytemark service:

container_name: Plausible-smtp
    hostname: mail
    image: bytemark/smtp
    restart: always
    environment:
      - MAILER_EMAIL=site@xyz.com
      - SMTP_HOST_ADDR=email-smtp.eu-central-1.amazonaws.com
      - SMTP_HOST_PORT=587
      - SMTP_USER_NAME=<username>
      - SMTP_USER_PWD=<password>
      - SMTP_HOST_SSL_ENABLED=false
   

To test, if i add a abc@gmail.com to site settings > people, the invitation gets sent, but the Sender address is hello@plausible.local and the message ALWAYS goes to the Spam folder. So the env var for MAILER_EMAIL doesn't work. Further to that, if i try to send the invite to mail@xyz.com - so another email address in the same custom domain of mine - the email doesn't get sent at all. I can only presume it's because my mail server refuses to allow hello@plausible.local to send mail on the xyz.com domain server. Why it works when i send to a gmail address i don't know - but fixing the MAILER_EMAIL issue seems to be the first step.

Should i use SMTP_HOST_SSL_ENABLED=true and port 465? All my other Amazon SES relays work with 587 so i stuck with that. The messages to the Gmail recipients are all tagged TLS standard encryption; so shouldn't that mean setting SMTP_HOST_SSL_ENABLED to 'true' and changing the port number? 

I've also tried the other format of env var that i've seen , which uses RELAY_HOST, RELAY_PORT etc, but same result; Gmail recipient email are sent, but only to Spam, and domain email recipient email are not sent, so that doesn't seem to change anything. 

I've struggled with this for days and really need some help.

**Comment by ruslandoga** [comment]
👋 @wheeleran 

It seems like your env vars are set on the `bytemark/smtp` container and not on `plausible/analytics` so they are no-op.

**Reply by ruslandoga** [reply]
Also note that `bytemark/smtp` is unnecessary if you use an SMTP relay like Amazon SES.

**Reply by ruslandoga** [reply]
`SMTP_HOST_SSL_ENABLED` controls whether the connection to SMTP server is opened in plain TCP or TLS. The latter only works on port 465 (according to the https://datatracker.ietf.org/doc/html/rfc8314). And on ports 25 and 587 (submission) `STARTTLS` is used to upgrade a plain TCP connection to TLS.

**Reply by wheeleran** [reply]
thanks for the replies! wow! so, to summarise: first, move the smtp env vars to the plausible service env vars, not bytemark. second, you're suggesting remove the bytemark service (and corresponding depends_on: mail) totally if i use amazon SES? does that mean i user RELAY_HOST etc, or SMTP_HOST_ADDR? third, SMTP_HOST_SSL_ENABLED: false in port 587, and SMTP_HOST_SSL_ENABLED: true if port 465. do i have all this right?

**Reply by ruslandoga** [reply]
I don't think `RELAY_HOST` is recognized by Plausible. The env vars that are used to configure the mailer are listed [in the docs.](https://plausible.io/docs/self-hosting-configuration#mailersmtp-setup)

**Reply by ruslandoga** [reply]
So in your setup, I think it'd be something like:

```
MAILER_EMAIL=site@xyz.com
SMTP_HOST_ADDR=email-smtp.eu-central-1.amazonaws.com
SMTP_HOST_PORT=587
SMTP_USER_NAME=...aws_ses_username...
SMTP_USER_PWD=...aws_ses_password...
```

`SMTP_HOST_SSL_ENABLED` is false by [default.](https://github.com/plausible/analytics/blob/16c4b9e3eb991399ec1c000d02364a75fed06a28/config/runtime.exs#L349-L361)

**Reply by wheeleran** [reply]
and move all that to the env vars for plausible, and out of bytemark? do you suggest i then remove the bytemark container completely? i haven't ever seen a docker-compose like that.

**Comment by ruslandoga** [comment]
Yes, if you use plausible-conf.env, that’s where you’d put these env vars. And yes, with that setup (aws ses) you can remove bytemark container completely:)

**Reply by wheeleran** [reply]
I don't use a separate plausible-conf, i just stick everything in docker-compose yaml. same?

**Reply by ruslandoga** [reply]
Right, then just adding these env vars under `environment:` on `plausible/analytics` container should work :)

**Reply by wheeleran** [reply]
You, Sir/Madam, are a wonderful human being. It works! Everything, nothing to Spam, to all destination email addresses, with MAILER_EMAIL correct and working, with the bytemark container completely removed, and all mail env vars in the plausible service environment . Fantastic. Thank you SOOOOO much!

**Reply by wheeleran** [reply]
One question @ruslandoga, is there any way to name the MAILER_EMAIL? as in: 'John Smith <email@example.com>' I'm know it's been asked in the past, and tried, and failed, but has this been fixed?

**Reply by ruslandoga** [reply]
@wheeleran I wonder what happens if you try setting `MAILER_EMAIL` to something like `"John Smith <email@example.com>"`

Do you get 500 as in https://github.com/plausible/analytics/discussions/2349#discussioncomment-3906328 or do you get it working?

**Reply by wheeleran** [reply]
Tried that, and also ```""John Doe" <abc@example.com>"``` and no, nothing works, always 500 Oops. If there's no name field to pass to the email service (and their isn't) then anything we try to cram in to MAILER_EMAIL to fake it returns an error. We need a MAILER_NAME, it seems.

**Reply by ruslandoga** [reply]
Is there any error in the logs?

```console
$ cd hosting
$ docker compose logs plausible
```

**Reply by wheeleran** [reply]
Um, not done that before. I just get an "Attaching to Plausible" message then creation of databases and running of migrations. What now?

**Reply by wheeleran** [reply]
> Is there any error in the logs?
> 
> ```
> $ cd hosting
> $ docker compose logs plausible
> ```

@ruslandoga i searched further, found your reply to ticket #2349 from 2021 which was the same request, and there doesn't seem to be any way to fake it, so i opened a Feature Request which got moved to Discussion #2925. Can you help?

**Reply by ruslandoga** [reply]
Sure, I'll ask the core team about it on the next call! :)

**Reply by wheeleran** [reply]
thanks! looking forward to hearing what they say!

**Reply by ruslandoga** [reply]
@wheeleran can you please try out this container image [`ghcr.io/ruslandoga/analytics:add-mailer-name-fork`](https://github.com/ruslandoga/analytics/pkgs/container/analytics/95333761?tag=add-mailer-name-fork) that is built for https://github.com/ruslandoga/analytics/pull/105 (note that it's based on [`v2.0.0-rc.0`](https://github.com/plausible/analytics/discussions/2896))

It adds an optional `MAILER_NAME` environment variable which configures the mailer's "name".

Changes to `plausible-conf.env`

```diff
MAILER_EMAIL=plausible@self.hosted
+ MAILER_NAME=John
```

Changes to `docker-compose.yml`

```diff
plausible:
- image: plausible/analytics:v2.0.0-rc.0
+ image: ghcr.io/ruslandoga/analytics:add-mailer-name-fork
```

**Reply by ruslandoga** [reply]
I'm asking because I'm failing to properly verify it works in a live environment: SMTP ports are blocked on my self-hosted instance and I don't have access to AWS SES at the moment.

**Reply by wheeleran** [reply]
I'll get right to it! Can MAILER_NAME be 'John Doe?' Does it respect spaces, with or without " quotation marks? Or John+Doe?

**Reply by ruslandoga** [reply]
Thank you!

I think it'd be 

```
MAILER_NAME="John Doe"
```
(it might also work without quotes but I'm not sure as it probably depends on the shell)

**Reply by wheeleran** [reply]
OK, not going well. I'm getting a Head error "https://ghcr.io/v2/ruslandago/analytics/manifests/add-mailer-name-fork": denied.

**Reply by ruslandoga** [reply]
I've been able to verify it works with `smtp.gmail.com` as a relay using the SMTP adapter.

**Reply by ruslandoga** [reply]
@wheeleran what are you using for `image:` in `plausible:` service in `docker-compose.yml`?

**Reply by wheeleran** [reply]
what you gave me: ghcr.io/ruslandoga/analytics:add-mailer-name-fork. Hang on, let me try something.

**Reply by ruslandoga** [reply]
Here's what worked for me:

```yml
  plausible:
    # image: plausible/analytics:v2.0.0-rc.0
    image: ghcr.io/ruslandoga/analytics:add-mailer-name-fork
    restart: always
    # command: sh -c "sleep 10 && /entrypoint.sh db createdb && /entrypoint.sh db migrate && /entrypoint.sh run"
    command: sh -c "sleep 10 && /entrypoint.sh run"
    depends_on:
      - plausible_db
      - plausible_events_db
      - mail
    ports:
      - 127.0.0.1:8000:8000
    env_file:
      - plausible-conf.env
```

**Reply by wheeleran** [reply]
@ruslandoga hey it works! env var was MAILER_NAME=Plausible Analytics, no quotation marks. no other changes to docker-compose, no changes to command line. added a Person as Admin and sent invite. all correct!!!

my current command line is ```command: sh -c "sleep 10 && /entrypoint.sh db createdb && /entrypoint.sh db migrate && /entrypoint.sh run"```

**Reply by ruslandoga** [reply]
Great! Thank you for trying it out!

**Reply by wheeleran** [reply]
merge it!! great work, thank you!!

**Reply by wheeleran** [reply]
@ruslandoga one question, actually two: any chance to modify the email template to remove the beginning "[Plausible Analytics]" in the subject line? and could you possible have a look at my ticket nr #2912. I'm really struggling with it. got it working in Umami, but not Plausible, and i want to *leave* Umami but can't if i can't fix this...

**Reply by ruslandoga** [reply]
> any chance to modify the email template to remove the beginning "[Plausible Analytics]" in the subject line?

There have been discussions about making the email templates customisable, but I don't think it included the subject lines. I'll ask the core team on the next call about it.

> could you possible have a look at my ticket nr https://github.com/plausible/analytics/discussions/2912

I'm not familiar with the actual tracking logic, sorry. But I'll ask the core team about this.

**Reply by wheeleran** [reply]
ok, much appreciated. what happens now with the fork? i've not done this before...

**Reply by ruslandoga** [reply]
You can continue using that container image from the fork until `v2.0.0` is released with that PR merged in. I'll post here once `v2.0.0` is released.

**Reply by ruslandoga** [reply]
[`v2.0.0-rc.1`](https://github.com/plausible/analytics/discussions/2981) has been released and it includes `MAILER_NAME` PR.
